build_projects/dotnet-cli-build/WriteVersionToVersionsRepo.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if !SOURCE_BUILD
+using Microsoft.Build.Framework;
+using MSBuildUtilities = Microsoft.Build.Utilities;
+using Microsoft.DotNet.VersionTools.Automation;
+using Microsoft.DotNet.VersionTools.Automation.GitHubApi;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using ThreadingTasks= System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class WriteVersionToVersionsRepo : MSBuildUtilities.Task
+    {
+        [Required]
+        public string BranchName { get; set; }
+
+        [Required]
+        public string Name { get; set; }
+
+        [Required]
+        public string Version { get; set; }
+
+        [Required]
+        public string GitHubPassword { get; set; }
+
+        public string VersionsRepoPath { get; set; }
+        
+        public override bool Execute()
+        {
+            string versionsRepoPath = VersionsRepoPath ?? $"build-info/dotnet/core-sdk/{BranchName}";
```

Comment:
I'd suggest leaving off the `?? $"build-info/dotnet/core-sdk/{BranchName}"` and marking `VersionsRepoPath` required to remove the duplication.